music-rpc.ts

```diff
@@ -283,7 +283,7 @@ async function setActivity(rpc: Client) {
           });
         }
 
-        const query = `artist:${props.artist} track:${props.name}`;
+        const query = encodeURIComponent(`artist:${props.artist} track:${props.name}`);
         const spotifyUrl = encodeURI(
           `https://open.spotify.com/search/${query}?si`
```

Comment:
Can't we just only once `encodeURIComponent` the whole url?

Reply:
If I wrap the entire URL and the query, Deno throws an error:

```
Error: (4000) child "activity" fails because [child "buttons" fails because ["buttons" at position 1 fails because [child "url" fails because ["url" must be a valid uri]]]]
    at DiscordIPC.#read (https://deno.land/x/discord_rpc@0.3.2/src/conn.ts:180:11)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async https://deno.land/x/discord_rpc@0.3.2/src/conn.ts:49:11
```

That being said, only the query needs to be encoded here as the URL is already valid the way it is. I've updated the code to reflect this and it does work nicely 👍

Reply:
Fair point!